static-assets/components/cstudio-common/common-api.js

```diff
@@ -3087,6 +3087,38 @@ var nodeOpen = false,
         CStudioAuthoring.Operations.openUploadDialog(site, path, isUploadOverwrite, uploadCb, fileTypes);
       },
 
+      directUploadAsset: function(file, site, path, callback) {
+        let serviceUri = `${CStudioAuthoring.Service.createServiceUri(
+          '/api/1/services/api/1/content/write-content.json'
+        )}&${CStudioAuthoringContext.xsrfParameterName + '=' + CrafterCMSNext.util.auth.getRequestForgeryToken()}`;
+
+        serviceUri += `&contentType=folder&createFolders=true&draft=false&duplicate=false&path=${encodeURIComponent(
+          path
+        )}&site=editorial&unlock=true`;
```

Comment:
Shouldn't we be using `/studio/asset-upload` endpoint?

Reply:
On the next content service, there's a `uploadDataUrl` method. Perhaps we could use that one or refactor it into two methods so we can use it with both kinds of upload.